client/app/lib/components/table/TanStackTableBuilder/useTanStackTableBuilder.tsx

```diff
@@ -82,8 +86,25 @@ const useTanStackTableBuilder = <D extends object>(
   });
 
   const generateAndDownloadCsv = async (): Promise<void> => {
+    const downloadableColumns = (table.getHeaderGroups()[0]?.headers ?? [])
+      .map(({ column: { columnDef: { header } = {} } = {} }) => header)
+      .filter((header): header is string => isString(header))
+      .reduce<string[]>((acc, header, index) => {
+        if (getRealColumn(index)?.csvDownloadable ?? true) {
+          acc.push(header);
+        }
+        return acc;
+      }, []);
+
+    const headers = table.options.columns
+      .filter(({ id, header }) => id || header)
+      .map(({ id, header }) => (header || id) as string)
+      .filter(
+        (header) => downloadableColumns.includes(header) && header !== 'index',
```

Comment:
Must you really do a separate loop to get headers?  
Moreover `downloadableColumns.includes(header)` will cost you O(headers^2) for no good reason.